app/classifier/tasks/survey/translations.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import merge from 'lodash/merge';
+import translations from '../../../pages/project/translations';
+
+function SurveyTranslations(props) {
+  const { task } = props;
+  const taskStrings = translations.strings.workflow.tasks;
+  let translation = merge({}, task);
+
+  function explodeTranslationKey(translationKey, value) {
+    const translationKeys = translationKey.split('.');
+    const translationObject = {};
+    let temp = translationObject;
+    while (translationKeys.length) {
+      temp[translationKeys[0]] = (translationKeys.length === 1) ? value : {};
+      temp = temp[translationKeys[0]];
+      translationKeys.shift();
+    }
+    return translationObject;
+  }
+
+  Object.keys(taskStrings).map((translationKey) => {
+    const newTranslation = explodeTranslationKey(translationKey, taskStrings[translationKey]);
```

Comment:
@simoneduca is this easier to understand than the nested maps? I've added a function that explodes the translation task keys out to nested objects, which can be merged into the translation.

Reply:
Yeah! I mean, there's lots going on, but it's looking pretty neat now.

Reply:
Cheers! I've added tests, which hopefully clarify what this component does.